primitives/version/src/lib.rs

```diff
@@ -246,6 +328,17 @@ impl RuntimeVersion {
 	}
 }
 
+impl RuntimeVersion {
+	/// Returns state version to use for update.
+	///
+	/// For runtime with core api version less than 4,
+	/// V0 trie version will be applied to state.
+	/// Otherwhise, V1 trie version will be use.
+	pub fn state_version(&self) -> StateVersion {
+		(self.state_version as u8).try_into().expect("Invalid state version")
```

Comment:
I switch to using V1 (latest known version) on error.
Could try to return an Option, but it may be just pushing the panic out out of the crate, or would require consequent api changes (eg return Option<Header> in frame executive instead of Header...).

Reply:
Okay fine.